### PR TITLE
Implement create, change and delete event logic

### PR DESCRIPTION
**Pull request by DavMovsesyan2002**

### CODE REVIEW

File: src/pages/events/createEvent.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+const CreateEvent = () => {
+  console.log(99)
```
Comment (gor017) [inline src/pages/events/createEvent.tsx:4]: Remove console.log

File: src/redux/slices/events/middleware.ts [hunk 2]
```diff
@@ -53,6 +54,20 @@ const getSingleEvent = (id: string) => async (dispatch: AppDispatch) => {
   }
 }
 
+const createEvent = (formData: ICreateEventParams) => async (dispatch: AppDispatch) => {
+  try {
+    dispatch(setSingleEventLoading(true))
+
+    await API.events.createEvent(formData)
+
+    // dispatch(setSingleEventData(response.data.data))
```
Comment (gor017) [inline src/redux/slices/events/middleware.ts:63]: why comment? get data and set it in redux store, get it in the component and show it

File: src/types/reduxTypes/eventsStateTypes.ts [hunk 3]
```diff
@@ -24,3 +24,10 @@ export interface IEventsListProps {
 export interface IEventsListParams {
   page: number
 }
+
+export interface ICreateEventParams {
+  header: string
+  paragraph: string
+  imageDescription: string
+  imageURL: File | null
```
Comment (gor017) [inline src/types/reduxTypes/eventsStateTypes.ts:32]: should imageURL be file or string?

**Review by gor017 — CHANGES_REQUESTED** [review]

File: src/redux/slices/events/middleware.ts [hunk 4]
```diff
@@ -51,14 +56,62 @@ const getSingleEvent = (id: string) => async (dispatch: AppDispatch) => {
...
+
+const changeEvent = (formData: IChangeEventFormProps) => async (dispatch: AppDispatch) => {
+  try {
+    await API.events.changeEvent(formData)
+
+    dispatch(setChangeEventSubmitSuccess(true))
+  } catch (error) {
+    /* empty */
+  }
+}
```
Comment (ArthurO2002) [inline src/redux/slices/events/middleware.ts:102]: Why no loading?

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]

File: src/types/reduxTypes/eventsStateTypes.ts (hunk 3 above)
Comment (alexkocharyan01) [reply src/types/reduxTypes/eventsStateTypes.ts]: image must be or File or null for create

File: src/redux/slices/events/middleware.ts (hunk 2 above)
Comment (alexkocharyan01) [reply src/redux/slices/events/middleware.ts]: this is not get data this is call request for create events

File: src/pages/events/createEvent.tsx (hunk 1 above)
Comment (alexkocharyan01) [reply src/pages/events/createEvent.tsx]: done

File: src/redux/slices/events/middleware.ts (hunk 4 above)
Comment (ArthurO2002) [reply src/redux/slices/events/middleware.ts:103]: In that case let's set redirect logic here instead of keeping state for that

Comment (alexkocharyan01) [reply src/redux/slices/events/middleware.ts]: done

**Review by ArthurO2002 — APPROVED** [review]

**Pull request merged**